File: DSA/SAMAGRA_590014891/Day_96_Question_2.c
#include <stdio.h>
#include <stdbool.h>

bool subsetSum(int arr[], int n, int target) {
    bool dp[n+1][target+1];
    for(int i=0;i<=n;i++){
        for(int j=0;j<=target;j++){
            if(j==0) dp[i][j]=true;
            else if(i==0) dp[i][j]=false;
            else if(arr[i-1]<=j)
                dp[i][j]=dp[i-1][j] || dp[i-1][j-arr[i-1]];
            else
                dp[i][j]=dp[i-1][j];
        }
    }
    return dp[n][target];
}

int main() {
    int arr[]={3,34,4,12,5,2};
    int n=sizeof(arr)/sizeof(arr[0]);
    int target=9;
    if(subsetSum(arr,n,target)) printf("Yes\n");
    else printf("No\n");
    return 0;
}

File: DSA/LakshaKundra_590013237/Day_96_Question_2.py
def subset_sum(arr, target):
    n = len(arr)
    dp = [[False]*(target+1) for _ in range(n+1)]
    for i in range(n+1):
        dp[i][0] = True
    for i in range(1, n+1):
        for s in range(1, target+1):
            if arr[i-1] <= s:
                dp[i][s] = dp[i-1][s] or dp[i-1][s-arr[i-1]]
            else:
                dp[i][s] = dp[i-1][s]
    return "Yes" if dp[n][target] else "No"

print(subset_sum([3,34,4,12,5,2], 9))
print(subset_sum([1,2,3], 7))
